backends/xsmm/ceed-xsmm-tensor.c

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+#include <ceed.h>
+#include <ceed/backend.h>
+#include <libxsmm.h>
+
+#include "ceed-xsmm.h"
+
+//------------------------------------------------------------------------------
+// Tensor Contract Apply
+//------------------------------------------------------------------------------
+static int CeedTensorContractApply_Xsmm(CeedTensorContract contract, CeedInt A, CeedInt B, CeedInt C, CeedInt J, const CeedScalar *restrict t,
+                                        CeedTransposeMode t_mode, const CeedInt add, const CeedScalar *restrict u, CeedScalar *restrict v) {
+  Ceed ceed;
+  CeedCallBackend(CeedTensorContractGetCeed(contract, &ceed));
+
+  if (C == 1) {
+    // Build or query the required kernel
+    const int                  flags_t    = LIBXSMM_GEMM_FLAGS(!t_mode ? 'T' : 'N', 'N');
+    const int                  flags_ab   = (!add) ? LIBXSMM_GEMM_FLAG_BETA_0 : 0;
```

Comment:
small question - is there a named enum/macro (not sure which `LIBXSMM_GEMM_FLAG_BETA` is) for 0 here?

Reply:
This is from https://github.com/libxsmm/libxsmm/blob/main_stable/samples/hello/hello.c. I think the alternative would be to replace `0` by `LIBXSMM_BASIC_GEMM_FLAG_NONE`.

Reply:
My preference is to use words if we have them.

Reply:
Done: 684fdff.